env0/resource_environment_discovery_configuration.go

```diff
@@ -0,0 +1,349 @@
+package env0
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceEnvironmentDiscoveryConfiguration() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentDiscoveryConfigurationPut,
+		ReadContext:   resourceEnvironmentDiscoveryConfigurationGet,
+		UpdateContext: resourceEnvironmentDiscoveryConfigurationPut,
+		DeleteContext: resourceEnvironmentDiscoveryConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceEnvironmentDiscoveryConfigurationImport},
+
+		Description: "See https://docs.env0.com/docs/environment-discovery for additional details",
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "the project id",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"glob_pattern": {
+				Type:        schema.TypeString,
+				Description: "the environments glob pattern. Any match to this pattern will result in an Environment creation and plan",
+				Required:    true,
+			},
+			"repository": {
+				Type:        schema.TypeString,
+				Description: "the repository to run discovery on",
+				Required:    true,
+			},
+			"type": {
+				Type:             schema.TypeString,
+				Description:      "the infrastructure type use. Valid values: 'opentofu', 'terraform', 'terragrunt', 'workflow' (default: 'opentofu')",
+				Default:          "opentofu",
+				ValidateDiagFunc: NewStringInValidator([]string{"opentofu", "terraform", "terragrunt", "workflow"}),
+				Optional:         true,
+			},
+			"environment_placement": {
+				Type:             schema.TypeString,
+				Description:      "the environment placement strategy with the project (default: 'topProject')",
+				Default:          "topProject",
+				ValidateDiagFunc: NewStringInValidator([]string{"existingSubProject", "topProject"}),
+				Optional:         true,
+			},
+			"workspace_naming": {
+				Type:             schema.TypeString,
+				Description:      "the Workspace namimg strategy (default: 'default')",
+				Default:          "default",
+				ValidateDiagFunc: NewStringInValidator([]string{"default", "environmentName"}),
+				Optional:         true,
+			},
+			"auto_deploy_by_custom_glob": {
+				Type:        schema.TypeString,
+				Description: "if configured, deploy/plan on changes matching the given pattern (glob). Otherwise, deploy/plan on changes in directories matching the main glob pattern",
```

Comment:
Description is wrong, the "main" glob pattern isn't related to this glob pattern - so essentially it should be:
```suggestion
				Description: "If specified, deploy/plan on changes matching the given pattern (glob). Otherwise, deploy on template folder changes only",